lib/api/include/irods/ticketAPI.hpp

```diff
@@ -22,114 +22,132 @@ struct RsComm;
 struct RcComm;
 #endif // IRODS_TICKET_ADMINISTRATION_ENABLE_SERVER_SIDE_API
 
-enum ticket_type
-{
-    READ,
-    WRITE
-};
-
-// WIP: user_constraint, host_constraint, and group_constraint
-enum ticket_prop
-{
-    USES,
-    WRITE_FILE,
-    WRITE_BYTE
-};
-enum ticket_operation
-{
-    SET,
-    REMOVE
-};
-
-class constraint
+/**
+ * @brief Namespace to contain the API for tickets
+ *
+ */
+namespace irods::administration::ticket
 {
-private:
-    ticket_prop property;
-    ticket_operation operation;
-    bool as_admin;
-    int ticket_ID;
-    std::string_view ticket_name;
-
-public:
-    constraint(ticket_prop _prop, ticket_operation _operation, bool admin_access, std::string_view _ticket_name)
-    {
-        property = _prop;
-        operation = _operation;
-        as_admin = admin_access;
-        ticket_name = _ticket_name;
-    }
-    constraint(ticket_prop _prop, ticket_operation _operation, bool admin_access, int ticketID)
+    inline struct admin_tag
     {
-        property = _prop;
-        operation = _operation;
-        as_admin = admin_access;
-        ticket_ID = ticketID;
-    }
-    // constraint()
+    } admin;
 
-    void set_ticket_prop(ticket_prop _prop)
-    {
-        property = _prop;
-    }
-    void set_operation(ticket_operation _oper)
-    {
-        operation = _oper;
-    }
-    void set_admin_access(bool admin_access)
+    enum class ticket_type
     {
-        as_admin = admin_access;
-    }
-    void set_ticket_ID(int ticketID)
+        read,
+        write
+    };
+
+    enum class ticket_property
     {
-        ticket_ID = ticketID;
-    }
-    void set_ticket_name(std::string_view _ticket_name)
+        uses,
+        write_file,
+        write_byte
+    };
+    enum class ticket_operation
     {
-        ticket_name = _ticket_name;
-    }
+        set,
+        remove
+    };
 
-    ticket_prop get_ticket_prop()
-    {
-        return property;
-    }
-    ticket_operation get_ticket_operation()
+    namespace constraint
     {
-        return operation;
-    }
-    bool get_admin_access()
+        std::string_view user_string = "user";
+        std::string_view group_string = "group";
+        std::string_view host_string = "host";
+
+        struct constraint
+        {
+        private:
+            std::string_view _value;
+            std::string_view ticket_identifier;
+            std::string_view group_type;
```

Comment:
The data type of these variables assume that the strings will outlive the constraint type.

Also, the iRODS convention for private data members is that they always have a trailing underscore, not a leading underscore.

Reply:
Removed in the new commit